### ISSUE TITLE
Unimplemented code assertion during debug

### ISSUE THREAD

**Issue by zcourts** [flutter/flutter#99379]
## Steps to Reproduce

It's not easy to reproduce, it only happened during debug. The application sent a request to the server, the server hit a breakpoint and the application was left stuck with no response coming back for a few hours (more like 5).

No sample code to reproduce because I don't have a consistent way to do so, it doesn't always happen if it is left stuck waiting on a server response.

**Expected results:**  For the app not to crash

**Actual results:** 
```
[+106152297 ms] ../../third_party/dart/runtime/vm/timeline.cc: 664: error: unimplemented code
[   +7 ms] version=2.14.3 (stable) (Wed Sep 29 13:10:26 2021 +0200) on "macos_x64"
[        ] pid=55775, thread=37379, isolate_group=(nil)(0x0), isolate=(nil)(0x0)
[        ] isolate_instructions=0, vm_instructions=10db32f40
[   +5 ms]   pc 0x000000010dc91865 fp 0x0000000308a4a430 dart::Profiler::DumpStackTrace(void*)+0x85
[        ]   pc 0x000000010db33154 fp 0x0000000308a4a510 dart::Assert::Fail(char const*, ...)+0x84
[  +35 ms]   pc 0x000000010dd1bc29 fp 0x0000000308a4a570 dart::TimelineEvent::PrintJSON(dart::JSONStream*) const+0x409
[        ]   pc 0x000000010dd1e23b fp 0x0000000308a4a5b0 dart::TimelineEventBlock::PrintJSON(dart::JSONStream*) const+0x5b
[        ]   pc 0x000000010dcfff70 fp 0x0000000308a4a600 dart::ServiceEvent::PrintJSON(dart::JSONStream*) const+0x370
[        ]   pc 0x000000010dcee252 fp 0x0000000308a4a6e0 dart::Service::HandleEvent(dart::ServiceEvent*)+0x112
[        ]   pc 0x000000010dd1c996 fp 0x0000000308a4a830 dart::TimelineEventRecorder::ThreadBlockStartEvent()+0xd6
[        ]   pc 0x000000010dd1bcd2 fp 0x0000000308a4a850 dart::TimelineStream::StartEvent()+0x32
[        ]   pc 0x000000010df8325f fp 0x0000000308a4a8c0 Dart_TimelineEvent+0x4f
[        ]   pc 0x000000010daf885e fp 0x0000000308a4a8f0 std::__1::__function::__func<void (*)(char const*, long long, long long, Dart_Timeline_Event_Type, long, char const**, char const**), std::__1::allocator<void (*)(char const*, long long, long long, Dart_Timeline_Event_Type, long, char const**, char const**)>, void (char const*, long long, long long, Dart_Timeline_Event_Type, long, char const**, char const**)>::operator()(char const*&&, long long&&, long long&&, Dart_Timeline_Event_Type&&, long&&, char const**&&, char const**&&)+0x3e
[        ]   pc 0x000000010d33f64f fp 0x0000000308a4a960 fml::tracing::TraceEventEnd(char const*)+0xaf
[        ]   pc 0x000000010d856045 fp 0x0000000308a4aa00 flutter::PictureLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x195
[        ]   pc 0x000000010d852161 fp 0x0000000308a4aa60 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4ab20 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4ab80 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d851323 fp 0x0000000308a4ac00 flutter::ClipRectLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0xa3
[        ]   pc 0x000000010d852161 fp 0x0000000308a4ac60 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4ad20 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4ad80 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d851323 fp 0x0000000308a4ae00 flutter::ClipRectLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0xa3
[        ]   pc 0x000000010d852161 fp 0x0000000308a4ae60 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d851323 fp 0x0000000308a4aee0 flutter::ClipRectLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0xa3
[        ]   pc 0x000000010d852161 fp 0x0000000308a4af40 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4b000 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b060 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4b120 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b180 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d851323 fp 0x0000000308a4b200 flutter::ClipRectLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0xa3
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b260 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4b320 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b380 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d8556f3 fp 0x0000000308a4b420 flutter::PhysicalShapeLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x63
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b480 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4b540 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b5a0 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4b660 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852161 fp 0x0000000308a4b6c0 flutter::ContainerLayer::PrerollChildren(flutter::PrerollContext*, SkMatrix const&, SkRect*)+0x71
[        ]   pc 0x000000010d85756b fp 0x0000000308a4b780 flutter::TransformLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x14b
[        ]   pc 0x000000010d852061 fp 0x0000000308a4b7e0 flutter::ContainerLayer::Preroll(flutter::PrerollContext*, SkMatrix const&)+0x81
[        ]   pc 0x000000010d853e92 fp 0x0000000308a4b950 flutter::LayerTree::Preroll(flutter::CompositorContext::ScopedFrame&, bool)+0x1b2
[        ]   pc 0x000000010d8403c0 fp 0x0000000308a4bae0 flutter::CompositorContext::ScopedFrame::Raster(flutter::LayerTree&, bool)+0x50
[        ]   pc 0x000000010d86b7a2 fp 0x0000000308a4bca0 flutter::Rasterizer::DrawToSurface(flutter::FrameTimingsRecorder&, flutter::LayerTree&)+0x1e2
[        ]   pc 0x000000010d86c923 fp 0x0000000308a4bd20 flutter::Rasterizer::DoDraw(std::__1::unique_ptr<flutter::FrameTimingsRecorder, std::__1::default_delete<flutter::FrameTimingsRecorder> >, std::__1::unique_ptr<flutter::LayerTree, std::__1::default_delete<flutter::LayerTree> >)+0x73
[        ]   pc 0x000000010d86e359 fp 0x0000000308a4bd70 std::__1::__function::__func<flutter::Rasterizer::Draw(std::__1::unique_ptr<flutter::FrameTimingsRecorder, std::__1::default_delete<flutter::FrameTimingsRecorder> >, std::__1::shared_ptr<flutter::Pipeline<flutter::LayerTree> >, std::__1::function<bool (flutter::LayerTree&)>)::$_2, std::__1::allocator<flutter::Rasterizer::Draw(std::__1::unique_ptr<flutter::FrameTimingsRecorder, std::__1::default_delete<flutter::FrameTimingsRecorder> >, std::__1::shared_ptr<flutter::Pipeline<flutter::LayerTree> >, std::__1::function<bool (flutter::LayerTree&)>)::$_2>, void (std::__1::unique_ptr<flutter::LayerTree, std::__1::default_delete<flutter::LayerTree> >)>::operator()(std::__1::unique_ptr<flutter::LayerTree, std::__1::default_delete<flutter::LayerTree> >&&)+0xd9
[        ]   pc 0x000000010d86bef5 fp 0x0000000308a4bdc0 flutter::Pipeline<flutter::LayerTree>::Consume(std::__1::function<void (std::__1::unique_ptr<flutter::LayerTree, std::__1::default_delete<flutter::LayerTree> >)> const&)+0xc5
[        ]   pc 0x000000010d86bb65 fp 0x0000000308a4beb0 flutter::Rasterizer::Draw(std::__1::unique_ptr<flutter::FrameTimingsRecorder, std::__1::default_delete<flutter::FrameTimingsRecorder> >, std::__1::shared_ptr<flutter::Pipeline<flutter::LayerTree> >, std::__1::function<bool (flutter::LayerTree&)>)+0xc5
[        ]   pc 0x000000010d888cd4 fp 0x0000000308a4bf40 std::__1::__function::__func<fml::internal::CopyableLambda<flutter::Shell::OnAnimatorDraw(std::__1::shared_ptr<flutter::Pipeline<flutter::LayerTree> >, std::__1::unique_ptr<flutter::FrameTimingsRecorder, std::__1::default_delete<flutter::FrameTimingsRecorder> >)::$_14>, std::__1::allocator<fml::internal::CopyableLambda<flutter::Shell::OnAnimatorDraw(std::__1::shared_ptr<flutter::Pipeline<flutter::LayerTree> >, std::__1::unique_ptr<flutter::FrameTimingsRecorder, std::__1::default_delete<flutter::FrameTimingsRecorder> >)::$_14> >, void ()>::operator()()+0xe4
[        ]   pc 0x000000010d3381a0 fp 0x0000000308a4bff0 fml::MessageLoopImpl::FlushTasks(fml::FlushType)+0x110
[        ]   pc 0x000000010d34081c fp 0x0000000308a4c020 fml::MessageLoopDarwin::OnTimerFire(__CFRunLoopTimer*, fml::MessageLoopDarwin*)+0x2c
[   +4 ms]   pc 0x00007ff808df493c fp 0x0000000308a4c030 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__+0x14
[        ]   pc 0x00007ff808df442d fp 0x0000000308a4d0e0 __CFRunLoopDoTimer+0x3a3
[        ]   pc 0x00007ff808df3f8c fp 0x0000000308a4d130 __CFRunLoopDoTimers+0x133
[        ]   pc 0x00007ff808dda6a7 fp 0x0000000308a4de70 __CFRunLoopRun+0x7c9
[        ]   pc 0x00007ff808dd9809 fp 0x0000000308a4df10 CFRunLoopRunSpecific+0x237
[        ]   pc 0x000000010d340a7e fp 0x0000000308a4df50 fml::MessageLoopDarwin::Run()+0x8e
[        ]   pc 0x000000010d337fb4 fp 0x0000000308a4df70 fml::MessageLoopImpl::DoRun()+0x24
[        ]   pc 0x000000010d33e77a fp 0x0000000308a4dfb0 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, fml::Thread::Thread(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::$_0> >(void*)+0x10a
[        ]   pc 0x00007ff808d15514 fp 0x0000000308a4dfd0 _pthread_start+0x7d
[        ]   pc 0x00007ff808d1102f fp 0x0000000308a4dff0 thread_start+0xf
[        ] -- End of DumpStackTrace
[ +117 ms] Service protocol connection closed.
[   +1 ms] Lost connection to device.
[   +5 ms] DevFS: Deleting filesystem on the device (file:///var/folders/4p/jczkx4js0tj3rnv5gk3wn7880000gn/T/io.hypi.tink.tinker.tinker/tinker84sUhD/tinker/)
[{"id":33,"result":true}]
```

flutter 2.5.3 - 2021 Mac with M1 chip

```
➜  tinker git:(feature/HYPI-1214) ✗ flutter doctor -v
[✓] Flutter (Channel stable, 2.5.2, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.5.2 at /Users/zcourts/.flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (5 months ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/zcourts/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/zcourts/Library/Android/sdk/
    • ANDROID_SDK_ROOT = /Users/zcourts/Library/Android/sdk
    • Java binary at: /Users/zcourts/.sdkman/candidates/java/current/bin/java
    • Java version OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.9.1+1)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.2.1, Build version 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/zcourts/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/zcourts/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/zcourts/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7678000/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 61.2.2
    • Dart plugin version 203.8430

[☠] IntelliJ IDEA Ultimate Edition (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know about this issue at
      https://github.com/flutter/flutter/issues.
    ✗ FormatException: Unexpected extension byte (at offset 5)
    • #0      _Utf8Decoder.convertSingle (dart:convert-patch/convert_patch.dart:1788:7)
      #1      Utf8Decoder.convert (dart:convert/utf.dart:318:42)
      #2      InputStream.readString (package:archive/src/util/input_stream.dart:199:30)
      #3      new ZipDirectory.read (package:archive/src/zip/zip_directory.dart:40:30)
      #4      ZipDecoder.decodeBuffer (package:archive/src/zip_decoder.dart:19:30)
      #5      ZipDecoder.decodeBytes (package:archive/src/zip_decoder.dart:14:12)
      #6      IntelliJPlugins._findPluginXml (package:flutter_tools/src/intellij/intellij.dart:130:44)
      #7      IntelliJPlugins._readPackageVersion (package:flutter_tools/src/intellij/intellij.dart:141:40)
      #8      IntelliJPlugins.validatePackage (package:flutter_tools/src/intellij/intellij.dart:63:35)
      #9      IntelliJValidator.validate (package:flutter_tools/src/intellij/intellij_validator.dart:103:15)
      #10     asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:111:32)
      #11     asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:109:18)
      #12     _rootRun (dart:async/zone.dart:1428:13)
      #13     _CustomZone.run (dart:async/zone.dart:1328:19)
      #14     _runZoned (dart:async/zone.dart:1863:10)
      #15     runZonedGuarded (dart:async/zone.dart:1851:12)
      #16     runZoned (dart:async/zone.dart:1782:12)
      #17     asyncGuard (package:flutter_tools/src/base/async_guard.dart:109:3)
      #18     Doctor.startValidatorTasks (package:flutter_tools/src/doctor.dart:200:9)
      #19     Doctor.diagnose (package:flutter_tools/src/doctor.dart:304:47)
      #20     DoctorCommand.runCommand (package:flutter_tools/src/commands/doctor.dart:50:47)
      #21     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1274:12)
      <asynchronous suspension>
      #22     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1125:27)
      <asynchronous suspension>
      #23     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #24     CommandRunner.runCommand (package:args/command_runner.dart:209:13)
      <asynchronous suspension>
      #25     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
      <asynchronous suspension>
      #26     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #27     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
      <asynchronous suspension>
      #28     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
      <asynchronous suspension>
      #29     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #30     main (package:flutter_tools/executable.dart:92:3)
      <asynchronous suspension>


[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.1)
    • IntelliJ at /Users/zcourts/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/213.6461.79/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.3)
    • IntelliJ at /Users/zcourts/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/212.5457.46/IntelliJ IDEA.app
    • Flutter plugin version 59.0.4
    • Dart plugin version 212.4746.57

[✓] VS Code (version 1.64.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 98.0.4758.109

! Doctor found issues in 1 category.

```

**Comment by a-siva** [comment]
This sounds like a duplicate of https://github.com/flutter/flutter/issues/87460 and a fix for that landed here
https://dart-review.googlesource.com/c/sdk/+/219905

@zcourts could you try using the latest stable version of Flutter.

**Comment by bkonyi** [comment]
Yeah, this is 100% related to #87460. Upgrading to the latest Dart/Flutter will fix this.
